pkg/gadgets/trace/tcpdump/tracer/compiler/compiler_tcpdump.go

```diff
@@ -0,0 +1,91 @@
+// Copyright 2022 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build localgadget
+
+package compiler
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"os/exec"
+	"strconv"
+	"strings"
+
+	"github.com/google/gopacket/layers"
+	"golang.org/x/net/bpf"
+)
+
+func TcpdumpExprToBPF(filterExpr string, linkType layers.LinkType, snapLen int) ([]bpf.RawInstruction, error) {
+	if linkType != layers.LinkTypeEthernet {
+		return nil, errors.New("unsupported linkType")
+	}
+
+	// We'll use an installed tcpdump to compile the filter
```

Comment:
In general, I also think a dependency on a lib would be better - but this file is only used for local-gadget that we compile statically. If depending on the lib, this would mean local-gadget wouldn't start without a (compatible) libpcap version installed - compiling libpcap statically into local-gadget seems non-trivial and could also be problematic regarding licenses (I didn't check). That's why I think this is the better solution here. But yes, that means that even if you would only want to capture to a file that you'd need tcpdump installed. Maybe we can provide an option to read the compiled filter from a file so that you could compile it on another computer and just copy/paste the instructions to the target machine. We could also provide a couple of precompiled generic rules where we just replace some variables (like ports, ips), but I'm not sure how far we should go that route...

Reply:
> If depending on the lib, this would mean local-gadget wouldn't start without a (compatible) libpcap version installed

Do we have any special requirements on the libpcap version? I'd think that libpcap offers a quite stable API and that in most of the cases it's not a problem for us. 

> compiling libpcap statically into local-gadget seems non-trivial

I agree. Perhaps we can have a mix in local-gadget, most of the functionality is statically compiled in it, but some extra functionality could require to have installed some dependencies. I think we could use dlopen() to check at runtime if a given library is available. 

Anyway, this seems too much for this PR, I'm totally fine with the dependency on the tcpdump binary for now.

Reply:
> I think we could use dlopen() to check at runtime if a given library is available.

Yeah, true. That would mean we have to rewrite a couple of things from the go pcap library, though. I'll look into it later on, then.